### ISSUE TITLE
Huawei SmartAX Custom Model

### ISSUE THREAD

**Issue by jfosnight** [ytti/oxidized#1295]
We have a few devices that are running Huawei's SmartAX Operating System.  I have tried to create a custom model for these, but I've ran into some issues.

The main issue is that when you enter a command in the CLI, if the command has any additional options, it prompts the user for further input.

For example, running: `display version`
the device outputs: `{ <cr>|backplane<K>|frameid/slotid<S><Length 1-15>||<K> }:`

At this point, if you hit enter again, it will go ahead and output all the version information.  A similar thing happens with `display current-configuration`  (Also you have to run `enable` before you can run display `current-config`, which changes the prompt from '>' to '#', but I think I fixed that with some extra regex for the prompt.
 
Here was my attempt at a model after messing around with it for awhile.
```ruby
class SmartAX < Oxidized::Model
  # Huawei SmartAX

  prompt /^([\w.-]+[>#])$/
  comment '# '

  expect /^\{ <cr>.* \}\:$/ do
    send "\r\n"
  end

  cmd :secret do |cfg|
    cfg.gsub! /(pin verify (?:auto|)).*/, '\\1 <PIN hidden>'
    cfg.gsub! /(%\^%#.*%\^%#)/, '<secret hidden>'
    cfg
  end

  cmd :all do |cfg|
    cfg.each_line.to_a[1..-2].join
  end

  cfg :telnet do
    username /^login as: $/
    password /^User password:$/
  end

  cfg :telnet, :ssh do
#    post_login 'screen-length 0 temporary'
    pre_logout 'quit'
  end

   cmd 'display version' do |cfg|
     comment cfg
   end

end
```

Any guidance would be helpful.  I'm not familiar at all with Ruby, so the syntax is a bit confusing.  

Thanks

**Comment by jsynack** [comment]
Hello @jfosnight ,

(Full disclosure - I do not have access to one of those devices - these are just generic thoughts)

For your `display version` command - is there an option such as `all` or anything that would proceed with the command.  I admit that I find the Huawei site a little troublesome to find simple command reference docs.

If there is not a 'one shot' way of running the command - There are a few different ways to send extra information (such as a CR).  Some information can be found on the [Ruby API page](https://github.com/ytti/oxidized/blob/master/docs/Ruby-API.md) .

It is also good to see examples of other models that require some of this behavior as well.
For example - [The Procurve model](https://github.com/ytti/oxidized/blob/master/lib/oxidized/model/procurve.rb) has a few `expect` blocks to catch things and send extra sequences to continue to command. 

The [FortiOS model](https://github.com/ytti/oxidized/blob/master/lib/oxidized/model/fortios.rb) also an example of an `expect` block (in this case, the paginator on the device).

Since you are using an expect block already - how do the debug logs show up for your session? Does oxidized match and proceed to send the CR or stall waiting to match a regular expression?  Debugging of the sessions can be adjusted in the config file - some more information [here](https://github.com/ytti/oxidized/blob/master/docs/Configuration.md#debugging)

**Comment by jfosnight** [comment]
Sorry for the long delay, I got pulled away to other projects.

Anyways, thanks for the feedback.  You pointed me enough in the write direction that I figured out what I was missing.  I was able to get a new model mostly working.  It still leaves a few artifacts of the CLI behind in the config, but it mostly pulls a clean config and at least isn't hanging up on the additional prompts.

**Comment by laf** [comment]
@jfosnight Are you going to submit a pull request with this new model so others can benefit?

**Comment by jfosnight** [comment]
In its current state, I don't think that it could be published due to some of the CLI artifacts.  However, I would be more than happy to share the code if anybody else needs it.

The main two issues are: 
- `display version` displays the uptime, which will show up as a config change every time (which isn't good for versioning.)
  - I've seen other models that strip this out, so I may be able to add that in.
- `display current-config` leaves the cli prompt at the end of the output.
- `display current-config` also leaves some info at the top of the output.
  - basically every time you enter a command in the CLI, it echos back what the command was that you entered. 

This is echoed after every command:
```

  Command:
          display current-configuration
```
